
# HG changeset patch
# User Ingo Schmidt <is@openoffice.org>
# Date 1271408858 -7200
# Node ID bc7556cee857016f7def0f3780a9dfb222cb0dff
# Parent  19b8efb810452b843cf6242713a002a688c96f2a
native299 #i110901# improving mechanism for DEFAULT_TO_ENGLISH_FOR_PACKING

diff -r 19b8efb81045 -r bc7556cee857 solenv/bin/modules/installer/archivefiles.pm
--- solenv/bin/modules/installer/archivefiles.pm	Fri Apr 16 10:40:37 2010 +0200
+++ solenv/bin/modules/installer/archivefiles.pm	Fri Apr 16 11:07:38 2010 +0200
@@ -50,19 +50,27 @@
 	my $filename = "";
 	my $extension = "";
 	
-	if ( $oldname =~ /^\s*(.*)(\..*?)\s*$/ )	# files with extension
+	if ( $oldname =~ /en-US/ )	# files, that contain the language in the file name
 	{
-		$filename = $1;
-		$extension = $2;	
+		$newname = $oldname;
+		$newname =~ s/en-US/$onelanguage/;
 	}
-	else
+	else	# files, that do not contain the language in the file name
 	{
-		$filename = $oldname;
-		$extension = "";	
+		if ( $oldname =~ /^\s*(.*)(\..*?)\s*$/ )	# files with extension
+		{
+			$filename = $1;
+			$extension = $2;	
+		}
+		else
+		{
+			$filename = $oldname;
+			$extension = "";	
+		}
+	
+		$newname = $1 . "_" . $onelanguage . $2;
 	}
 	
-	$newname = $1 . "_" . $onelanguage . $2;
-	
 	return $newname;
 }
 

